### PR TITLE
API-91: extract product's fields from productFilter

### PR DESCRIPTION
**Pull request by skeleton**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

Extract product's fields from ProductFilter to be able to use this new Filter on EE

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        $originalData = $this->normalizer->normalize($product, 'standard');
+        $result = [];
+
+        foreach ($newFields as $field => $value) {
+            if (!in_array($field, $this->productFields)) {
+                throw new \LogicException(sprintf('Cannot filter value of field "%s"', $field));
+            }
+
+            $comparator = $this->comparatorRegistry->getFieldComparator($field);
+            $originalData = !isset($originalData[$field]) ? null : $originalData[$field];
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php]: `??`

File: src/Pim/Component/Catalog/Comparator/Filter/ProductFilter.php [hunk 2]
```diff
@@ -58,44 +64,25 @@ public function filter(ProductInterface $product, array $newProduct)
...
-     */
-    protected function compareField(array $originalValues, $field, $code)
-    {
-        $comparator = $this->comparatorRegistry->getFieldComparator($code);
-        $diff = $comparator->compare($field, $this->getOriginalField($originalValues, $code));
-
-        if (null !== $diff) {
-            return [$code => $diff];
-        }
+        $result = $this->mergeValueToResult($result, $this->productFieldFilter->filter($product, $fields));
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Comparator/Filter/ProductFilter.php]: I would split it in modified in two lines `modifiedFields= ...` and then  `$this->merge (to know the behavior of the filter).

File: src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Filter product's fields to have only updated or new values
+ *
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductFieldFilter implements ProductFilterInterface
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php]: ProductFieldFilter seems a little bit too generic. `ModifiedProductFieldFielter` ? I don't have a better idea for now.

File: src/Pim/Component/Catalog/Comparator/Filter/ProductFilter.php (hunk 2 above)
Comment (momoss) [inline src/Pim/Component/Catalog/Comparator/Filter/ProductFilter.php]: Should be more readable

File: src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php (hunk 3 above)
Comment (skeleton) [inline src/Pim/Component/Catalog/Comparator/Filter/ProductFieldFilter.php]: no, because it filter also nullable values

**Pull request closed without merging**